pyaerocom/io/read_emep.py

```diff
@@ -0,0 +1,304 @@
+
+
+    @property
+    def vars_provided(self):
+        """Variables provided by this dataset"""
+        return self._get_vars_provided()
+
+
+    def _get_ts_types(self):
+        ts_types = []
+        if self.data_dir:
+            files = os.listdir(self.data_dir)
+            for filename in files:
+                ts_types.append(ts_type_from_filename(filename))
+        elif self.filepath:
+            filename = self.filepath.split('/')[-1]
+            ts_types.append(ts_type_from_filename(filename))
+        return list(set(ts_types))
+
+    def _get_paths(self):
+        paths = []
+        if self.filepath:
+            paths = [self.filepath]
+        if self.data_dir:
+            pattern = os.path.join(self.data_dir, 'Base_*.nc')
+            paths = glob.glob(pattern)
+        return paths
+
+    def _get_vars_provided(self):
+        variables = None
+        data_vars = set()
+        for filepath in self._get_paths():
+            data = xr.open_dataset(filepath)
+            data_vars.update(set(data.keys()))
+        emep_vars = set(get_emep_variables().values())
+        available = data_vars.intersection(emep_vars)
+        inv_emep = {v: k for k, v in get_emep_variables().items()}
+        avail_aero = [ inv_emep[var] for var in available]
+        variables = sorted(avail_aero)
+        return variables
+
+    @property
+    def data_id(self):
+        """
+        Data ID of dataset
+        """
+        return self._data_id
+
+    @data_id.setter
+    def data_id(self, val):
+        """
+        """
+        if not isinstance(val, str):
+            val = None
+        self._data_id = val
+
+    def __str__(self):
+        s = 'Reader: ReadEMEP\n'
+        s += "Available frequencies: {}\n".format(self.ts_types)
+        s += "Available variables: {}\n".format(self.vars_provided)
+        return s
+
+
+    def read_var(self, var_name, start=None, stop=None,
+                 ts_type=None):
+        """Read EMEP variable, rename to Aerocom naming and return GriddedData object"""
+
+        if start or stop:
+            raise NotImplementedError('Currently ReadEMEP only reads from files containing one year of data.')
+
+        var_map = get_emep_variables()
+
+        aliases = get_aliases(var_name)
+        if len(aliases) == 1 and aliases[0] in var_map:
+            var_name = aliases[0]
+
+        # Find path to file based on ts_type
+        filepath = ''
+        if self.data_dir:
+            filename = ''
+            if ts_type == 'monthly':
+                filename = 'Base_month.nc'
+            elif ts_type == 'daily':
+                filename = 'Base_day.nc'
+            elif ts_type == 'yearly':
+                filename = 'Base_fullrun.nc'
+            filepath = os.path.join(self.data_dir, filename)
+        elif self.filepath:
+            filepath = self.filepath
+
+        if var_name in AUX_REQUIRES:
+            temp_cubes = []
+            for aux_var in AUX_REQUIRES[var_name]:
```

Comment:
Please add `self.` for class header attributes.

Reply:
Done.